x-pack/plugins/alerting/server/alerts_client/alerts_client.ts

```diff
@@ -1470,6 +1512,7 @@ export class AlertsClient {
       actions: rawAlert.actions
         ? this.injectReferencesIntoActions(id, rawAlert.actions, references || [])
         : [],
+      params: this.injectReferencesIntoParams(id, ruleType, params, references || []) as Params,
```

Comment:
does `params` here always come from saved object ? if not, it should first be migrated (`updateToLatest` function needs to be implemented)

also, saved object migrations definition should include any migrations defined on rule types

Reply:
@ppisljar Yes, the params here should come directly from the saved object.

This particular issue was for getting the extract/inject hooks for saved object references working. We would be tackling the work for migrations in a separate issue. Since we don't currently have rule type defined migrations, I think it is ok to leave that for another issue...do you agree?